unleash-yggdrasil/src/state.rs

```diff
@@ -15,6 +15,32 @@ pub struct InnerContext {
     pub properties: Option<HashMap<String, String>>,
 }
 
+pub struct EnrichedContext {
```

Comment:
I have mixed feelings about having a secondary context struct with enriched properties. It looks a bit ugly but it buys us something very, very nice - the language specific SDK code is now shielded from changes in the enriched context which means internal implementations and extensions can happen without impacting the langauge SDK code